### PR TITLE
Remove networkx as dependency

### PR DESCRIPTION
**Pull request by jcrist**
I also want to swap out pydot for https://github.com/xflr6/graphviz, as it supports python 3, and has a better API (in my opinion).

Still needs tests - I didn't add these yet because I'm not sure if we'll still use pydot, or switch.

### CODE REVIEW

**Comment by mrocklin** [comment]
Switching sounds fine to me

**Comment by mrocklin** [comment]
Adding pydot to all travis builds kills py34 builds

**Comment by jcrist** [comment]
This is ready for review. This removes networkx and pydot as dependencies, and now relies on https://github.com/xflr6/graphviz instead. Visualizations now work in python 2.6, 2.7, 3.3, and 3.4. Code is also smaller.

File: .travis.yml [hunk 1]
```diff
@@ -6,6 +6,10 @@ python:
     - "3.3"
     - "3.4"
 
+addons:
+    apt:
+        packages:
+        - graphviz
```
Comment (mrocklin) [inline .travis.yml]: I had no idea that you could do this.

File: dask/dot.py [hunk 2]
```diff
@@ -1,99 +1,74 @@
...
     except AttributeError:
         return 'func'
 
 
-def to_networkx(d, data_attributes=None, function_attributes=None):
+def name(x):
+    try:
+        return str(hash(x))
+    except TypeError:
+        return str(hash(str(x)))
```
Comment (mrocklin) [inline dask/dot.py]: Hashes make thiings harder to debug.  Thoughts on continuuing to return x if x is hashable?

File: dask/dot.py [hunk 3]
```diff
@@ -1,99 +1,74 @@
...
+                    g.node(dep_name, label=str(dep), shape='box',
+                           **data_attributes.get(dep, {}))
+                g.edge(dep_name, func_name)
+        elif ishashable(v) and v in dsk:
+            g.edge(name(v), k_name)
+    return g
 
-    os.system('dot -Tpdf %s.dot -o %s.pdf' % (filename, filename))
-    os.system('dot -Tpng %s.dot -o %s.png' % (filename, filename))
-    print("Writing graph to %s.pdf" % filename)
```
Comment (mrocklin) [inline dask/dot.py]: Definitely agree with removing this.

File: dask/dot.py (hunk 2 above)
Comment (jcrist) [inline dask/dot.py]: I'd rather not. This is the name of the node of the graph, not the label. This identifier is only used to refer to nodes in the dot file. Passing in regular python objects means that they're printed, and used as the identifier. This meant that some names could be huge expressions, and there was no uniformity. Hashes are opaque, but uniform, and the labels for each node are still descriptive.

Comment (mrocklin) [inline dask/dot.py]: OK.  I suppose my concerns would be mollified if we were to improve node labeling, particularly for complex tasks like 

    (f, x, (g, y, 10))

Naming this as just `f` loses a lot of the meaning.  

I wonder if we could effectively names these nodes as follows:

    f(-, g(-, 10))

This is related to #413 I guess and could be useful in a few places.

**Comment by mrocklin** [comment]
This looks good to me

File: dask/dot.py (hunk 2 above)
Comment (jcrist) [inline dask/dot.py]: I plan to unite these in my next PR - all the tokenizing/naming stuff should be reused. OK to merge now, and fix in another PR?

Comment (mrocklin) [inline dask/dot.py]: Merge now and fix in another PR sounds good.

**Comment by mrocklin** [comment]
It looks like this no longer displays literals as in 

    {'x': 1}

gets named as 

    'x'

and not as 

    'x = 1'

Not sure if we wanted the old behavior or not.

**Comment by jcrist** [comment]
I intentionally changed this, but could be persuaded to put it back. For dask array, sometimes we'd get huge arrays printed in a node. The names seemed to be clearer than the actual data.

**Comment by mrocklin** [comment]
Yeah, I definitely remember the giant bricks of numpy or pandas values.  My guess is that we want the literals when the length of their string representation is less than 50 or so.  Thoughts?

**Comment by mrocklin** [comment]
This came up when I was looking at reusing this image from your optimization doc page

**Comment by jcrist** [comment]
Hmmmm, I'm not sure how I feel about being inconsistent with printing, but that could be doable. Having just the names of the keys doesn't make that less readable to me though; I actually prefer having smaller nodes. I don't feel strongly about this though. Is this something you want?

**Comment by mrocklin** [comment]
I don't have a strong opinion here either.  Lets leave it as is until something comes up.

**Pull request merged**